src/machine/machine_atmega328p.go

```diff
@@ -467,3 +468,96 @@ var SPI0 = SPI{
 	sdo:  PB3,
 	sdi:  PB4,
 	cs:   PB2}
+
+// Pin Change Interrupts
+var (
+	pinCallbacks [3][8]func(Pin)
+	interrupts   [3]*interrupt.Interrupt
+	previous     [3]uint8
+)
+
+func handlePCINTInterrupts(intr uint8, port *volatile.Register8) {
+	current := port.Get()
+	for i := uint8(0); i < 8; i++ {
+		if (current>>i)&0x01 == 0x01 && pinCallbacks[intr][i] != nil {
+			pinCallbacks[intr][i](Pin(i * (intr + 1)))
+		}
+	}
+	previous[intr] = current
+}
+
+func handlePCINT0Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT0-avr.IRQ_PCINT0, avr.PINB)
+}
+
+func handlePCINT1Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT1-avr.IRQ_PCINT0, avr.PINC)
+}
+
+func handlePCINT2Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT2-avr.IRQ_PCINT0, avr.PIND)
+}
+
+func SetInterrupt(pin Pin, callback func(Pin)) (err error) {
+	if callback == nil {
+		// Disable this pin interrupt
+		return errors.New("Not implemented")
+	} else {
```

Comment:
`errors.New` causes a heap allocation, which is not a good idea in this code. Instead, please define this error in a global.
Or better yet: add it to src/machine/machine.go.

There are two other `errors.New` calls below that should be fixed in a similar way (but perhaps not all by defining them in src/machine/machine.go).

Also: please indent this code according to https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.

Reply:
Ah, good point. Addressed.